File: go/bird-watcher/bird_watcher.go
// Package birdwatcher implements the [Bird Watcher] exercise.
//
// [Bird Watcher]: https://exercism.org/tracks/go/exercises/bird-watcher
package birdwatcher

// TotalBirdCount returns the total bird count by summing the individual days'
// counts.
func TotalBirdCount(birdsPerDay []int) int {
	var count int
	for _, birds := range birdsPerDay {
		count += birds
	}

	return count
}

// BirdsInWeek returns the bird count by summing only the counts from the given
// week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	start := (week - 1) * 7
	return TotalBirdCount(birdsPerDay[start : start+7])
}

// FixBirdCountLog returns the bird counts after correcting for the one bird
// that was hiding in a far corner of the garden every second day.
func FixBirdCountLog(birdsPerDay []int) []int {
	for i := range birdsPerDay {
		if i%2 == 0 {
			birdsPerDay[i]++
		}
	}

	return birdsPerDay
}

File: clojure/bird-watcher/src/bird_watcher.clj
(ns bird-watcher)

(def last-week
  [0 2 5 3 7 8 4])

(defn today [birds]
  (last birds))

(defn inc-bird [birds]
  (update birds (- (count birds) 1) inc))

(defn day-without-birds? [birds]
  (boolean (some zero? birds)))

(defn n-days-count [birds n]
  (reduce + (take n birds)))

(defn busy-days [birds]
  (count (filter #(>= % 5) birds)))

(defn odd-week? [birds]
  (= [1 0 1 0 1 0 1]
     birds))
